src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,960 @@
+		? "FlyoutThemeTouchMinWidth" : "FlyoutThemeMinWidth"];
+				var visibleBounds = this.LayoutSlotWithMarginsAndAlignments;
+				// DXamlCore.GetCurrent().GetVisibleContentBoundsForElement(GetHandle(), &visibleBounds);
+
+				templateSettingsConcrete.FlyoutContentMinWidth = Math.Min(visibleBounds.Width, flyoutContentMinWidth);
+			}
+
+			double maxItemKeyboardAcceleratorTextWidth = 0;
+
+			IObservableVector<object> menuItems = Items;
+
+			// MenuFlyoutItem's alignment changes based on the layout of other MenuFlyoutItems.
+			// This check looks through all MenuFlyoutItems in our items source and finds the max
+			// keyboard accelerator label width, which affects the look of other items.
+			var nCount = (menuItems as ItemCollection).Size;
+			for (var i = 0; i < nCount; ++i)
+			{
+				MenuFlyoutItemBase item;
+				MenuFlyoutItem itemAsMenuItem;
+
+				item = Items[i] as MenuFlyoutItemBase;
+
+				itemAsMenuItem = item as MenuFlyoutItem;
+				if (itemAsMenuItem != null)
+				{
+					var desiredSize = (itemAsMenuItem as MenuFlyoutItem).GetKeyboardAcceleratorTextDesiredSize();
+					var desiredWidth = desiredSize.Width;
+					if (desiredWidth > maxItemKeyboardAcceleratorTextWidth)
+					{
+						maxItemKeyboardAcceleratorTextWidth = desiredWidth;
+					}
+				}
+			}
+
+			for (var i = 0; i<nCount; ++i)
+			{
+				MenuFlyoutItemBase item;
+				MenuFlyoutItem itemAsMenuItem;
+
+				item = Items[i] as MenuFlyoutItemBase;
+
+				itemAsMenuItem = item as MenuFlyoutItem;
+				if (itemAsMenuItem != null)
+				{
+					(itemAsMenuItem as MenuFlyoutItem).UpdateTemplateSettings(maxItemKeyboardAcceleratorTextWidth);
+				}
+			}
+		}
+
+		protected override void OnGotFocus(RoutedEventArgs pArgs)
+		{
+			var focusState = FocusState;
+			if (m_iFocusedIndex == -1 &&
+				focusState != FocusState.Unfocused)
+			{
+				// The MenuFlyoutPresenter gets focused for the first time right after it is opened.
+				// In this case we want to send focus to the first focusable item.
+				CycleFocus(true /* shouldCycleDown */, focusState);
+			}
+			else if (focusState == FocusState.Unfocused)
+			{
+				// A child element got focus, so make sure we keep m_iFocusedIndex in sync
+				// with it.
+				var focusedElement = FocusManager.GetFocusedElement() as DependencyObject;
+
+				// Since GotFocus is an async event, the focused element could be null if we got it
+				// after the popup closes, which clears focus.
+				if (focusedElement != null)
+				{
+					var focusedElementIndex = IndexFromContainer(focusedElement);
+					if (focusedElementIndex != -1)
+					{
+						m_iFocusedIndex = focusedElementIndex;
+					}
+				}
+			}
+
+
+		}
+
+		void EnsureInitialFocusIndex()
+		{
+			if (m_iFocusedIndex == -1)
+			{
+				var focusedElement = FocusManager.GetFocusedElement();
+
+				if (this != focusedElement)
+				{
+					IObservableVector<object> menuItems = Items;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'menuItems'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)